lib/rules/no-array-constructor.js

```diff
@@ -98,6 +137,10 @@ module.exports = {
 				"Array",
 			);
 
+			const isSpreadCase =
+				node.arguments.length === 1 &&
+				node.arguments[0].type === "SpreadElement";
```

Comment:
Code like `Array(5, ...foo)` could create an array with 5 empty items if `foo` has zero length, so I think it's safe to autofix only if there are zero arguments or at least two non-spread arguments.

Also, we can move this calculation inside `if (variable && variable.identifiers.length === 0) {` because it's not necessary otherwise.